[Storage] Support Immunablity Policy in Official powershell release

blueww opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

To support WORM (Immunablity Policy) in powershell official release, the PR for storage branch is: https://github.com/Azure/azure-powershell/pull/6151
The design pass review on
https://github.com/Azure/azure-powershell-pr/wiki/%5BAzureRM.Storage%5D-ImmutabilityPolicy

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

maddieclayton commented on src/ResourceManager/Storage/Commands.Management.Storage.Test/ScenarioTests/StorageBlobTests.cs:
```diff
@@ -0,0 +1,52 @@
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestStorageBlobContainer()
+        {
+            TestController.NewInstance.RunPsTest("Test-StorageBlobContainer");
```
If you pull in latest from preview you will see that a new parameter has been added - this should be
```
TestController.NewInstance.RunPsTest("Test-StorageBlobContainer", interceptor);
```

And you will need to create a local XunitTracingInterceptor and add that at line 28 (see other test cs files for example).

maddieclayton commented on src/ResourceManager/Storage/Commands.Management.Storage.Test/Commands.Management.Storage.Test.csproj:
```diff
@@ -59,18 +60,22 @@
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
-    <None Include="SessionRecords\Microsoft.Azure.Commands.Management.Storage.Test.ScenarioTests.StorageAccountTests\TestGetAzureStorageAccount.json">
```
You need to add this back in to get the test to pass

blueww replied:
change the file by adding all *.json in the session record folder.

maddieclayton commented:
@blueww Please also fix the static analysis errors: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/7020/artifact/src/Package/SignatureIssues.csv/*view*/

blueww commented:
@maddieclayton 
I have fixed the Unittest and the StaticAnalysis error.
BTW, why we force the cmdlets must have output, and need 'PassThru' for the bool output.

blueww commented:
@maddieclayton 

The .net build failed in the last. Seems not related with my code change, would you please help to look?

> 23:00:46 <<<<
> 23:00:46 
> 23:00:46 ERROR
> 23:00:46 
> 23:00:46 Credential scanner match found.
> 23:00:46 Details can be found in the CredScanIssues.csv file.
> 23:00:46 Path to the file: artifact/src/Package/CredentialScannerOutput/CredScanIssues.csv
> 23:00:46 For false positives, follow the instructions on this doc: https://github.com/Azure/adx-documentation-pr/blob/master/engineering/credscan-ps.md
> 23:00:46 For more information, please read the 'How should I respond to reports of incorrectly handled secrets?' section 
> 23:00:46 of this doc: https://microsoft.sharepoint.com/teams/AzureSecurityCompliance/Security/SitePages/SecretsManagement.aspx.
> 23:00:46 
> 23:00:46 >>>>
> 23:00:46 ==== CredScanner: Exit script
> 

The netcore build failed since following error, but I don't find network project reference Microsoft.Azure.Management.Storage, how whould I fix it？(Actually, I have upgrade all netcore project reference Microsoft.Azure.Management.Storage 7.1 to reference  8.1)


[xUnit.net 00:00:16.13]     Commands.Network.Test.ScenarioTests.ApplicationGatewayTests.TestApplicationGatewayCRUD [FAIL]

Failed   Commands.Network.Test.ScenarioTests.ApplicationGatewayTests.TestApplicationGatewayCRUD

Error Message:

 System.Management.Automation.RuntimeException : Test failed due to a non-empty error stream. First error: PowerShell Error Record: Could not load file or assembly 'Microsoft.Azure.Management.Storage, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

MiYanni commented:
@azuresdkci Retest this please

blueww commented:
@MiYanni 
The netcore build still fail, I thinks it might related with SRP SDK is referenced in azure-powershell-common:
https://github.com/Azure/azure-powershell-common/blob/ef926a1e40814b79c7d14f0a8afc235c93523428/src/Storage.Management/Storage.Management.Netcore.csproj#L39 
https://github.com/Azure/azure-powershell-common/blob/ef926a1e40814b79c7d14f0a8afc235c93523428/src/Storage/Storage.Netcore.csproj#L41

I will try to upgrade SRPSDK in azure-powershell-common (still use the number 1.0.96), the XSCL will still not 9.0 (not upgrade to 9.3) as this PR doesn't include XSCL upgrade code.

MiYanni commented:
@blueww I merged the OAuth PR. So, this PR needs to be updated (conflicts resolved) so that it can build properly.

blueww commented:
@MiYanni 

I have fixed the merge conflict, and fixed the cmdlet name for new cmdlets.
But the netcore build still fail as following error, any idea?


>  System.Management.Automation.CommandNotFoundException : The term 'New-AzureRmStorageContainer' is not recognized as the name of a cmdlet, function, script file, or operable program.

MiYanni commented:
@blueww There is an issue with the prefixing for Storage cmdlet names right now. @maddieclayton will look into making the fix for your branch here.

blueww commented:
Thanks for the helping!
When can it been fixed?
Let me know if need any help from me.

Br,
Wei

在 2018年9月7日，上午1:53，Michael Yanni <notifications@github.com<mailto:notifications@github.com>> 写道：


@blueww<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fblueww&data=02%7C01%7Cweiwei%40microsoft.com%7C7a0ec88b78d644c4048e08d61421b69b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636718532358149281&sdata=%2FVv8BVV5jwysEZFKASXCJPK3bErinRWRCIWwCvhSm0U%3D&reserved=0> There is an issue with the prefixing for Storage cmdlet names right now. @maddieclayton<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmaddieclayton&data=02%7C01%7Cweiwei%40microsoft.com%7C7a0ec88b78d644c4048e08d61421b69b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636718532358149281&sdata=IKSbeW0jYuMLDz92ffi9qBB6uzupt7kpJ%2Bk3Xj2GD30%3D&reserved=0> will look into making the fix for your branch here.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-powershell%2Fpull%2F7091%23issuecomment-419184462&data=02%7C01%7Cweiwei%40microsoft.com%7C7a0ec88b78d644c4048e08d61421b69b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636718532358149281&sdata=Nm4%2FKz6qPzY1sshjzcIHz16KSX5hV4%2BFLHJB8gNQrA8%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGmvcRR9GK23YRkZs8KVYyx_RFOvTIUWks5uYWEygaJpZM4WRQXt&data=02%7C01%7Cweiwei%40microsoft.com%7C7a0ec88b78d644c4048e08d61421b69b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636718532358149281&sdata=%2BAMIFW6bspZTcY1FHLfU4yW5yRiuEmqwqgnGYzbQSRY%3D&reserved=0>.

blueww commented:
On Demand: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1083/
ps-sign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/1187/
psnetsign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/161/

MiYanni commented:
@blueww Ah. psnetsign ran on worker 78s. It might not have the latest PowerShellGet. Maddie fixed that issue with this PR, https://github.com/Azure/azure-powershell/pull/7154, but I think that worker isn't updated properly. But, everything else looks good. I'll review this PR in a few hours from now.

MiYanni commented:
psnetsign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/176/

MiYanni commented:
demand: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1087/
ps-sign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/1216/
psnetsign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/178/

blueww commented:
@MiYanni , @maddieclayton 
Thanks very much for all the work to make the PR merged!
